conda_smithy/cli.py

```diff
@@ -179,14 +179,34 @@ def __init__(self, parser):
         scp = self.subcommand_parser
         scp.add_argument("--feedstock_directory", default=os.getcwd(),
                          help="The directory of the feedstock git repository.")
+        scp.add_argument("-c", "--commit", nargs='?', choices=["edit", "auto"], const="edit",
```

Comment:
Thanks. I'm pretty happy with it myself. 😄 This was inspired by my experiences contributing to Homebrew. 

Yeah, was thinking the same things last night when I wrote it. My main considerations for these choices are as follows.

First, not all re-renderings are simple feedstock updates. So it is nice to be able to explain what the re-rendering was intended to accomplish. While one can always amend after, IMHO it is nice to have one command to do the whole thing.

Second, the default edit mode feels a little nicer from the novice `git`/`conda-smithy` user perspective. Mainly because it is obvious a commit is happening. Also it lists the files changed and shows the diff of what is being committed in the editor. This can be great for catching something that shouldn't have been committed ( handy even for an experienced `git` user 😉 ). Though not being a novice with either, I may be a bad judge of what helps.

That all being said, I'm not terribly attached to which one is default. So if `auto` is a better fit, am happy to change it.